### PR TITLE
Refactoring : Extract method getBackboneAtomArray changes to reduce the cyclomatic complexity

### PR DESCRIPTION
**Pull request by xcalibur91**
The method getBackboneAtomArray had a cyclomatic complexity of 22, hence refactored the method was to optimize the code.
This refactoring can be qualified as extract method refactoring.
The logic and the output after refactoring remain the same.

### CODE REVIEW

**Review by josemduarte — CHANGES_REQUESTED** [review]
Thanks! nice improvements. See one comment below.

File: biojava-structure/src/main/java/org/biojava/nbio/structure/StructureTools.java [hunk 1]
```diff
@@ -1128,65 +1128,39 @@ public static Atom[] getRepresentativeAtomArray(Structure s) {
 	 * @return an Atom[] array
 	 */
 	public static Atom[] getBackboneAtomArray(Structure s) {
-
 		List<Atom> atoms = new ArrayList<>();
+		Set<String> nucleotideAtomNames = new HashSet<>(Arrays.asList(C1_ATOM_NAME, C2_ATOM_NAME, C3_ATOM_NAME, C4_ATOM_NAME, O2_ATOM_NAME, O3_ATOM_NAME, O4_ATOM_NAME, O5_ATOM_NAME, OP1_ATOM_NAME, OP2_ATOM_NAME, P_ATOM_NAME));
+		Set<String> aminoAcidAtomNames = new HashSet<>(Arrays.asList(CA_ATOM_NAME, C_ATOM_NAME, N_ATOM_NAME, O_ATOM_NAME));
```
[the remark this answers is not in the document]
Comment (xcalibur91) [reply biojava-structure/src/main/java/org/biojava/nbio/structure/StructureTools.java:1133]: Sure! I'll get it done

[the remark this answers is not in the document]
Comment (xcalibur91) [reply biojava-structure/src/main/java/org/biojava/nbio/structure/StructureTools.java]: have done the needful.

**Review by josemduarte — APPROVED** [review]
LGTM thanks!

**Pull request merged**